File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.py
def solution(s):
    strange_list = []
    # 순회하면서 복사하는데 짝수면 대문자, 홀수면 소문자로
    # 만약 공백이 나오면 pass하고 문자가 나오면 다시 짝수부터 세기
    # 각 단어별로 나누기
    pair = True
    for c in s:
        if c == ' ': 
            pair = True
            strange_list.append(' ')
            continue
        if pair:
            strange_list.append(c.upper())
        else:
            strange_list.append(c.lower())
        pair = not pair
    return("".join(strange_list))

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        /*
        문자열 순회하면서 공백이 아니면 짝홀수 시작, 공백이면 단어 끝
        */
        StringBuilder sb = new StringBuilder();
        boolean pair = true;
        for(char ch: s.toCharArray()){
            if(ch == ' '){
                pair = true;
                sb.append(ch);
                continue;
            }
            if(pair){
                if('a' <= ch && ch <= 'z'){
                    sb.append((char)(ch - 32));
                }
                else{
                    sb.append(ch);
                }
                pair = false;
            }
            else{
                if('A' <= ch && ch <= 'Z'){
                    sb.append((char)(ch + 32));
                }
                else{
                    sb.append(ch);
                }
                pair = true;
            }
        }
        return sb.toString();
    }
}